great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -565,6 +727,83 @@ def _pandas_column_map_condition_values(
         return list(domain_values[: result_format["partial_unexpected_count"]])
 
 
+def _pandas_multicolumn_map_condition_values(
+    cls,
+    execution_engine: PandasExecutionEngine,
+    metric_domain_kwargs: Dict,
+    metric_value_kwargs: Dict,
+    metrics: Dict[str, Any],
+    **kwargs,
+):
+    """Return values from the specified domain that match the map-style metric in the metrics dictionary."""
+    (
+        boolean_mapped_unexpected_values,
+        compute_domain_kwargs,
+        accessor_domain_kwargs,
+    ) = metrics["unexpected_condition"]
+    df, _, _ = execution_engine.get_compute_domain(
+        domain_kwargs=compute_domain_kwargs,
+        domain_type=MetricDomainTypes.IDENTITY.value,
+    )
+
+    if "column_list" not in accessor_domain_kwargs:
+        raise ValueError(
+            "_pandas_multicolumn_map_condition_values requires column_list in accessor_domain_kwargs"
```

Comment:
Small tweak here: the user won't know what "accessor_domain_kwargs" are, because they're generated internally based on the domain_kwargs.

As we're continuing to refine this, I'd want to figure out what the correct point for this validation is, since this is denoted an internal method, we should be able to have the input validation for the domain happen at the public method call. However, since these functions are being called by GE internally when built from potentially "upstream" metrics, it makes sense to me to put a user-facing error and check here.

```suggestion
            "no 'column_list' found in provided metric_domain_kwargs, but it is required for a multicolumn map metric (_pandas_multicolumn_map_condition_values)"
```

Reply:
@jcampbell These error messages were patterned after the ones for the single-column domain keys, such as in the following:
```
    if "column" not in accessor_domain_kwargs:
        raise ValueError(
            "_pandas_column_map_condition_values requires a column in accessor_domain_kwargs"
        )
```

I agree with your direction, especially this part (emphasized):
```
these functions are being called by GE internally when built from potentially "upstream" metrics
```
because I added the column verification against `metrics["table.columns"]` as part of the `Checkpoints` project.

Hence, how about we change the error messages consistently for `column` and `column_list` keys in a dedicated PR?

Reply:
That works for me.